gravitee-am-ui/src/app/settings/domains/domains.component.ts

```diff
@@ -16,26 +16,55 @@
 import { Component, OnInit } from "@angular/core";
 import { AppConfig } from "../../../config/app.config";
 import { ActivatedRoute } from "@angular/router";
-import { EnvironmentService } from "../../services/environment.service";
+import { DomainService } from "../../services/domain.service";
 
 @Component({
   selector: "app-domains",
   templateUrl: "./domains.component.html",
   styleUrls: ["./domains.component.scss"],
 })
 export class DomainsComponent implements OnInit {
+  private searchValue: string;
+  page: any = {};
   title = AppConfig.settings.portalTitle;
   version = AppConfig.settings.version;
   domains = [];
   currentEnvironment: any;
 
-  constructor(private route: ActivatedRoute) {}
+  constructor(private route: ActivatedRoute,
+              private domainService: DomainService) {
+    this.page.pageNumber = 0;
+    this.page.size = 10;
+  }
 
   ngOnInit() {
-    this.domains = this.route.snapshot.data["domains"];
+    const pagedApps = this.route.snapshot.data['domains'];
+    this.domains = pagedApps.data;
+    this.page.totalElements = pagedApps.totalCount
+  }
+
+  onSearch(event) {
+    this.searchValue = event.target.value;
+    this.loadApps();
+  }
+
+  loadApps() {
+    const findApps = (this.searchValue) ?
+      this.domainService.search('*' + this.searchValue + '*', this.page.pageNumber, this.page.size) :
+      this.domainService.findByEnvironment(this.page.pageNumber, this.page.size);
+
+    findApps.subscribe(pagedApps => {
+      this.page.totalElements = pagedApps.totalCount;
+      this.domains = pagedApps.data;
+    });
+  }
```

Comment:
Domains instead of Apps

- loadApps -> loadDomains
- findApps -> findDomains
- pagedApps -> pagedDomains